### ISSUE TITLE
Feature: Resize windows without entering adjustment mode

### ISSUE THREAD

**Issue by rwjack** [pop-os/shell#1519]
I thought this would be a highly requested feature, but I haven't found any trace of it, other than something similar: https://github.com/pop-os/shell/issues/1400

Resizing windows without entering adjustment mode could be implemented in the same way as the above issue - without default shortcuts, so users have the option to chose.

I think adjustment mode looks nice, but it has a major drawback: those extra 2 steps, entering and exiting. It would go a long way having the option to resize windows with a single button press.

**Comment by joelostblom** [comment]
I think this would be a really useful feature as well and it is similar to how more established window managers such as i3 handles resizing. Coming from i3 myself this is one of the things I miss the most and it does make PopOS feel clunky to use everytime I need to resize a window. I currently use the most to resize because it is faster, but it has other issues such as #709

**Comment by mmstick** [comment]
The future COSMIC desktop will support this: https://github.com/pop-os/cosmic-epoch/issues/36

**Comment by rwjack** [comment]
This makes me so happy about the future of pop-os!

**Comment by dominsights** [comment]
> The future COSMIC desktop will support this: [pop-os/cosmic-epoch#36](https://github.com/pop-os/cosmic-epoch/issues/36)

Awesome.

**Comment by MostHated** [comment]
Is there, or will be there a way to set window size directly via an API? I am wanting to be able to script it to run based on various inputs or triggers, ex window title, current monitor, time of day, etc.

**Comment by krmbzds** [comment]
As a Pop!_OS user, I don't care about tiling, the most important features for me are:
- ability to move/resize floating windows using only the keyboard (HJKL)
- having window sizes and positions being remembered

I personally prefer adjustment mode, I like vim and hate holding down modifier keys.

Here's my subjective Pros & Cons list FWIW:

| direct hotkey |  adjustment mode |
|---|---|
| single step (faster) |  multi-step (slower) |
| immediate feedback |  ability to preview before taking action, slow but friendly, might be better if you don't move/resize windows too often |
| need to hold down a modifier while moving windows | one hand operation after entering adjustment mode,  no need to hold down modifiers to move windows |
| unnatural to switch between moving and resizing | easy to switch between moving and resizing during adjustment since it requires a single modifier key - I do this a lot and it feels natural to me at this point |
| need to override multiple system shortcuts | only requires one shortcut for adjustment mode - doesn't interfere with system shortcuts |

And the biggest con of doing away with adjustment mode is that it will break the habits of existing users. There might be other Pros like simplifying logic in the code but I'm only evaluating from an end-user perspective not a developer.

**Comment by mmstick** [comment]
@MostHated If you're asking about cosmic-comp, you can ask at https://github.com/pop-os/cosmic-comp
pop-os/shell is only for tracking issues for the GNOME tiling extension in Pop!_OS today.

**Comment by MostHated** [comment]
No, I was asking about Gnome, if it was possible to get the same end result as using the hotkeys and entering 'Adjustment Mode' (ex, Super + Enter, then Shift + arrow, etc) via script.

**Comment by joaomendoncaa** [comment]
Is this still not possible with popshell?

**Comment by mmstick** [comment]
No, all feature development is on COSMIC.

**Comment by EnriqueWood** [comment]
I desperately need this, it would significantly improve my workflow. Currently, I'm finding it quite frustrating to switch in and out of Adjustment mode just to resize windows.

I'm working on a bash script that aims to simplify this process, and I've had some relative success so far. However, I could definitely use some help to refine it further. Here is it: [resize.sh](https://github.com/EnriqueWood/gnome-window-resizer/blob/main/resize.sh).

If anyone is interested in trying it out or contributing to its improvement, I would greatly appreciate it.

**Comment by joaomendoncaa** [comment]
> I desperately need this, it would significantly improve my workflow. Currently, I'm finding it quite frustrating to switch in and out of Adjustment mode just to resize windows.
> 
> I'm working on a bash script that aims to simplify this process, and I've had some relative success so far. However, I could definitely use some help to refine it further. Here is it: [resize.sh](https://github.com/EnriqueWood/gnome-window-resizer/blob/main/resize.sh).
> 
> If anyone is interested in trying it out or contributing to its improvement, I would greatly appreciate it.

I've hoped into alternative desktop enviroments and/or window managers/compositors in the mean time.

Extremely excited for COSMIC, which will fix all of our issues with popos-shell and give us even more super powers and ease-of-use. Should come out in pop-os stable this year! 🎉
